com.microsoft.mrtk.windowsspeech/Subsystems/TextToSpeech/WindowsTextToSpeechSubsystem.cs

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.MixedReality.Toolkit.Subsystems;
+using System;
+using System.Runtime.InteropServices;
+using UnityEngine;
+using UnityEngine.Scripting;
+#if WINDOWS_UWP
+using System.Threading.Tasks;
+using Windows.Foundation;
+using Windows.Media.SpeechSynthesis;
+using Windows.Storage.Streams;
+#endif // WINDOWS_UWP
+
+namespace Microsoft.MixedReality.Toolkit.Speech.Windows
+{
+    [Preserve]
+    [MRTKSubsystem(
+        Name = "com.microsoft.mixedreality.speech",
+        DisplayName = "Windows Text-To-Speech Subsystem",
+        Author = "Microsoft",
+        ProviderType = typeof(WindowsTextToSpeechSubsystemProvider),
+        SubsystemTypeOverride = typeof(WindowsTextToSpeechSubsystem),
+        ConfigType = typeof(BaseSubsystemConfig))]
+    public class WindowsTextToSpeechSubsystem : TextToSpeechSubsystem
+    {
+        [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.SubsystemRegistration)]
+        static void Register()
+        {
+            // Fetch subsystem metadata from the attribute.
+            var cinfo = XRSubsystemHelpers.ConstructCinfo<WindowsTextToSpeechSubsystem, TextToSpeechSubsystemCinfo>();
+
+            if (!WindowsTextToSpeechSubsystem.Register(cinfo))
+            {
+                Debug.LogError($"Failed to register the {cinfo.Name} subsystem.");
+            }
+        }
+
+        [Preserve]
+        class WindowsTextToSpeechSubsystemProvider : Provider
+        {
+            public WindowsTextToSpeechSubsystemProvider() : base()
+            { }
+
+            public override void Destroy()
+            {
+#if WINDOWS_UWP
+                if (synthesizer != null)
+                {
+                    synthesizer.Dispose();
+                    synthesizer = null;
+                }
+#endif // WINDOWS_UWP
+            }
+
+            #region ITextToSpeechSubsystem implementation
+
+            /// <inheritdoc/>
+            public override bool TrySpeak(string phrase, AudioSource audioSource)
```

Comment:
I'd recommend some human-readable warning or error if you try and call this when not on Windows. It looks like you've properly ifdeffed everything for it to fail elegantly when not on Windows, but maybe just an early-out warning like "You've called TrySpeak on the Windows-only implementation of the TTS subsystem on a non-Windows platform" or something?

Reply:
sounds good. i had something like that in there early in the process and must have removed it by accident

Reply:
added in 129cebc